### DISCUSSION TITLE
课程6 relax.Var报错

### DISCUSSION THREAD

**Discussion by ShuRaymond** [mlc-ai/mlc-zh#192] [Q&A] [answered]
A = relax.Var("A", (128, 128), relax.DynTensorType(2, "float32"))
B = relax.Var("B", (128, 128), relax.DynTensorType(2, "float32"))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-47-18a0b9fae867>](https://localhost:8080/#) in <cell line: 1>()
----> 1 A = relax.Var("A", (128, 128), relax.DynTensorType(2, "float32"))
      2 B = relax.Var("B", (128, 128), relax.DynTensorType(2, "float32"))

[/usr/local/lib/python3.10/dist-packages/tvm/relax/expr.py](https://localhost:8080/#) in __init__(self, name_hint, struct_info, span)
    414             struct_info = tvm.runtime.convert_to_object(struct_info)
    415             if not isinstance(struct_info, StructInfo):
--> 416                 raise TypeError(
    417                     "struct_info needs to be an instance of StructInfo. "
    418                     "If you attempt to pass in shape, "

TypeError: struct_info needs to be an instance of StructInfo. If you attempt to pass in shape, use relax.TensorStructInfo(shape, dtype).

**Comment by hanquanjushi** [comment]
excuse me , have you already solve it ?

**Reply by Jvanlo** [reply, marked as answer]
relax.Var("A", relax.TensorStructInfo((128, 128), "float32"))

**Reply by hanquanjushi** [reply]
thank you so much!
